app/models/review.rb

```diff
@@ -11,4 +11,22 @@ class Review < ActiveRecord::Base
 
   validates :rating, presence: true,
              inclusion: { in: RATINGS }
+
+  def total_votes
+    votes.sum(:score)
+  end
+
+  def increment_vote(user)
+    fetch_vote(user).increment
+  end
```

Comment:
It's validated within the vote model with methods can_upvote? can_downvote? which check the score that user has submitted previously for this question